### DISCUSSION TITLE
Prompt for validation before deleting representation in the interface ?

### DISCUSSION THREAD

**Discussion by Gorgious56** [IfcOpenShell/IfcOpenShell#7031] [BlenderBIM User Interface]
While switching between representations I often miss-click and delete representations which in some cases leads to a bug. (eg deleting a representation while it is active)

![Animation3](https://github.com/user-attachments/assets/43031de9-004e-49ca-bcae-cb442070b3aa)

```
  File "C:\Users\\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\bonsai\bim\module\geometry\operator.py", line 366, in _execute
    core.remove_representation(
  File "C:\Users\\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\bonsai\core\geometry.py", line 180, in remove_representation
    geometry.delete_data(data)
  File "C:\Users\\AppData\Roaming\Blender Foundation\Blender\4.2\extensions\.local\lib\python3.11\site-packages\bonsai\tool\geometry.py", line 109, in delete_data
    bpy.data.meshes.remove(data)
ReferenceError: StructRNA of type Mesh has been removed
```

So this report is twofold :

- Prevent user from deleting active representation
- Maybe add a prompt before deleting a representation ?
   Alternatively, use SHIFT + click for force deleting a representation ?

**Comment by theoryshaw** [comment]
Undo, doesn't work here?

**Comment by Gorgious56** [comment]
I try not using UNDO if I don't absolutely have to since it was and still is in my experience n° 1 cause for file desync. 

It seems after a bit of testing that in this case it is safe but the first point still stands IMO, it should either be forbidden to delete the active representation or be silently dealt with. Maybe it's just a UX problem and the delete button is too close to the "Make Active" button and it's a problem of buttery fingers ><

Also confirm-before-delete is the default for Blender in a lot of other instances, I think it would make sense to have it there, too.

**Comment by falken10vdl** [comment]
@Gorgious56 , @theoryshaw  The code currently is robust to avoid this problem. I have however spotted that the "is_active" flag was not being updated for mapped representations. Please take a look to https://github.com/IfcOpenShell/IfcOpenShell/pull/7009
Cheers!

**Comment by falken10vdl** [comment]
@Gorgious56 @theoryshaw Are you ok to clise this one? The PR (and the robust code slready in place) is already merged. This should not be a problem.
If something is removed by mistake it will not throw an error and one can UNDO safely.

**Comment by Gorgious56** [comment]
Sure I cannot test it right now but I'll take your word. Cheers
